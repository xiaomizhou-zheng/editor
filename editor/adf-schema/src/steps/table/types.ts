import { Node as ProseMirrorNode } from 'prosemirror-model';
import { SideEffects, SideEffectsJSON } from './utils/side-effects/types';
import { SortOrder } from './constants';

export type ColumnInfo = Map<number, CellStep>;

export interface AddColumnStepInfo {
  cells: CellStep[];
  sideEffects?: SideEffects;
}

/**
 * Information about the cell
 * from: where the cell will be added removed
 * to: where the original cell ends
 * newCell?: the content of the new cells if it's added
 */
export interface CellStep {
  from: number; // beginning of cell at column
  to: number; // end of cell at column

  // Used to add a cell with a specific content. Useful for reverse a delete add column
  newCell?: ProseMirrorNode;

  mergeWith?: number; // Represent the cell that has to do the merge operation
}

export interface CellStepJson {
  from: number;
  to: number; // end of cell at column

  // Used to add a cell with a specific content. Useful for reverse a delete add column
  newCell?: { [key: string]: any }; // ToJson type of ProseMirrorNode.toJson()
  mergeWith?: number; // Represent the cell that has to do the merge operation
}

export interface AddColumnStepJson {
  stepType: 'ak-add-column';
  tablePos: number;
  cells: CellStepJson[];
  sideEffects?: SideEffectsJSON;
  isDelete: boolean;
}

export interface TableColumnOrdering {
  columnIndex: number;
  order: SortOrder;
}
