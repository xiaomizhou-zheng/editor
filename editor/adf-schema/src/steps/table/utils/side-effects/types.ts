import { Node as ProseMirrorNode } from 'prosemirror-model';

export interface TableSideEffect {
  from: number;
  to: number;
  node: ProseMirrorNode;
}

export interface RowSideEffect {
  from: number;
  to: number;
  rowNode: ProseMirrorNode;
}

export interface TableSideEffectJSON {
  from: number;
  to: number;
  node: { [key: string]: any }; // ToJson type of ProseMirrorNode.toJson()
}

export interface RowSideEffectJSON {
  from: number;
  to: number;
  rowNode: { [key: string]: any }; // ToJson type of ProseMirrorNode.toJson()
}

export type SideEffects = {
  table?: TableSideEffect;
  rows?: RowSideEffect[];
};
export type SideEffectsJSON = {
  table?: TableSideEffectJSON;
  rows?: RowSideEffectJSON[];
};
