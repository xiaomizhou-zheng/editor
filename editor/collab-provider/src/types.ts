import type { Transaction } from 'prosemirror-state';
import type { Step } from 'prosemirror-transform';
import type {
  CollabParticipant,
  CollabEventTelepointerData as EditorCollabTelepointerData,
  CollabEventConnectionData as EditorCollabConnetedData,
  CollabEventInitData as EditorCollabInitData,
  CollabEventRemoteData as EditorCollabData,
  CollabEventPresenceData as EditorCollabPresenceData,
} from '@atlaskit/editor-common/collab';
import type { AnalyticsWebClient } from '@atlaskit/analytics-listeners';
import type { Manager } from 'socket.io-client';
import type { DisconnectReason } from './disconnected-reason-mapper';

export interface Storage {
  get(key: string): Promise<string>;
  set(key: string, value: string): Promise<void>;
  delete(key: string): Promise<void>;
}

export interface Config {
  url: string;
  documentAri: string;
  lifecycle?: Lifecycle;
  storage?: Storage;
  // ESS-1009 Allow to opt-in into 404 response
  need404?: boolean;
  createSocket: (
    path: string,
    auth?: (cb: (data: object) => void) => void,
    productInfo?: ProductInformation,
  ) => Socket;
  analyticsClient?: AnalyticsWebClient;
  getUser?(
    userId: string,
  ): Promise<
    Pick<CollabParticipant, 'avatar' | 'email' | 'name'> & { userId: string }
  >;
  permissionTokenRefresh?: () => Promise<string>;
  productInfo?: ProductInformation;
}

interface SimpleEventEmitter {
  on(event: string, fn: Function): SimpleEventEmitter;
}

export interface Socket extends SimpleEventEmitter {
  id: string;
  connect(): Socket;
  emit(event: string, ...args: any[]): Socket;
  close(): Socket;
  io?: Manager;
}

export type LifecycleEvents = 'save' | 'restore';
export type EventHandler = () => void;

export interface Lifecycle {
  on(event: LifecycleEvents, handler: EventHandler): void;
}

export type CollabConnectedPayload = EditorCollabConnetedData;
export interface CollabDisconnectedPayload {
  reason: DisconnectReason;
  sid: string;
}
export interface CollabErrorPayload {
  status: number;
  code: string;
  message: string;
  reason?: string;
}
export interface CollabInitPayload extends EditorCollabInitData {
  doc: any;
  version: number;
  userId?: string;
  metadata?: Metadata;
}

export interface CollabDataPayload extends EditorCollabData {
  version: number;
  json: StepJson[];
  userIds: string[];
}

export type CollabTelepointerPayload = EditorCollabTelepointerData;
export type CollabPresencePayload = EditorCollabPresenceData;
export type CollabMetadataPayload = Metadata;
export type CollabLocalStepsPayload = {
  steps: Step[];
};

export interface CollabEvents {
  'metadata:changed': CollabMetadataPayload;
  init: CollabInitPayload;
  connected: CollabConnectedPayload;
  disconnected: CollabDisconnectedPayload;
  data: CollabDataPayload;
  telepointer: CollabTelepointerPayload;
  presence: CollabPresencePayload;
  'local-steps': CollabLocalStepsPayload;
  error: CollabErrorPayload;
  entity: any;
}

// Channel
export interface Metadata {
  [key: string]: string | number | boolean;
}

export type InitPayload = {
  doc: any;
  version: number;
  userId?: string;
  metadata?: Metadata;
};

export type PresencePayload = {
  sessionId: string;
  userId: string;
  clientId: string;
  timestamp: number;
};

export type TelepointerPayload = PresencePayload & {
  selection: {
    type: 'textSelection' | 'nodeSelection';
    anchor: number;
    head: number;
  };
};

export type StepJson = {
  from?: number;
  to?: number;
  stepType?: string;
  clientId: string;
  userId: string;
};

export type StepsPayload = {
  version: number;
  steps: StepJson[];
};

export type ErrorPayload = {
  message: string;
  data?: {
    status: number;
    code?: string;
    meta?: string | { description: string; reason?: string };
  };
};

export type ChannelEvent = {
  connected: {
    sid: string;
    initialized: boolean;
  };
  init: InitPayload;
  reconnected: null;
  'presence:joined': PresencePayload;
  presence: PresencePayload;
  'participant:left': PresencePayload;
  'participant:telepointer': TelepointerPayload;
  'participant:updated': PresencePayload;
  'steps:commit': StepsPayload & { userId: string };
  'steps:added': StepsPayload;
  'metadata:changed': Metadata;
  error: ErrorPayload;
  disconnect: { reason: string };
};

export interface CatchupResponse {
  doc?: string;
  version?: number;
  stepMaps?: any[];
  metadata?: Metadata;
}

// Catchup
export interface CatchupOptions {
  getCurrentPmVersion: () => number;
  fetchCatchup: (fromVersion: number) => Promise<CatchupResponse>;
  fitlerQueue: (condition: (stepsPayload: StepsPayload) => boolean) => void;
  getUnconfirmedSteps: () =>
    | {
        version: number;
        steps: Step<any>[];
        clientID: string | number;
        origins: Transaction<any>[];
      }
    | null
    | undefined;
  updateDocumentWithMetadata: ({
    doc,
    version,
    metadata,
    reserveCursor,
  }: CollabInitPayload) => void;
  applyLocalsteps: (steps: Step[]) => void;
}

export type ProductInformation = {
  product: string;
  subProduct?: string;
};
