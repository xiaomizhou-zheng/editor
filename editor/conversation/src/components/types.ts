import { ProviderFactory } from '@atlaskit/editor-common/provider-factory';
import { Comment as CommentType, User } from '../model';
import { Editor as AkEditor, EditorProps } from '@atlaskit/editor-core';
import { CommentAction as AkCommentAction } from '@atlaskit/comment';
import { SuccessHandler } from '../internal/actions';
import { EventData } from '../internal/analytics';

export type SendAnalyticsEvent = (eventData: EventData) => void;

export type RenderEditorWithComments = (
  Editor: typeof AkEditor,
  props: EditorProps,
  comment?: CommentType,
) => JSX.Element;

/**
 * Props which are passed down from the parent Conversation/Comment
 */
export interface SharedProps {
  user?: User;
  comments?: CommentType[];
  /* Optionally restricts visual nesting of comments to a maximum level. */
  maxCommentNesting?: number;

  // Dispatch
  onAddComment?: (
    conversationId: string,
    parentId: string,
    value: any,
    localId?: string,
    onSuccess?: SuccessHandler,
  ) => void;
  onUpdateComment?: (
    conversationId: string,
    commentId: string,
    value: any,
    onSuccess?: SuccessHandler,
  ) => void;
  onDeleteComment?: (
    conversationId: string,
    commentId: string,
    onSuccess?: SuccessHandler,
  ) => void;
  onRevertComment?: (conversationId: string, commentId: string) => void;
  onCancelComment?: (conversationId: string, commentId: string) => void;
  onCancel?: () => void;
  onHighlightComment?: (
    event: React.MouseEvent<HTMLAnchorElement>,
    commentId: string,
  ) => void;
  onEditorOpen?: () => void;
  onEditorClose?: () => void;
  onEditorChange?: (
    isLocal: boolean,
    value: any,
    conversationId: string,
    commentId: string | undefined,
    meta: any,
    objectId: string,
    containerId?: string,
  ) => void;

  // Provider
  dataProviders?: ProviderFactory;

  // Event Hooks
  onUserClick?: (user: User) => void;
  onRetry?: (localId?: string) => void;
  onCommentPermalinkClick?: (
    event: React.MouseEvent<HTMLAnchorElement>,
    commentId: string,
  ) => void;

  // Editor
  renderEditor?: RenderEditorWithComments;

  objectId?: string;
  containerId?: string;

  isHighlighted?: boolean;
  placeholder?: string;
  disableScrollTo?: boolean;
  allowFeedbackAndHelpButtons?: boolean;
  sendAnalyticsEvent: SendAnalyticsEvent;

  portal?: HTMLElement;

  renderAdditionalCommentActions?: (
    CommentAction: typeof AkCommentAction,
    comment: CommentType,
  ) => JSX.Element[];
  renderAfterComment?: (comment: CommentType) => JSX.Element;
}
