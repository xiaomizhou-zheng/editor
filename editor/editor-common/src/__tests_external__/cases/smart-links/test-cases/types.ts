import { EditorTestCaseOpts } from '../../types';

export interface SmartLinkTestCaseOpts extends EditorTestCaseOpts {
  url: string;
}
