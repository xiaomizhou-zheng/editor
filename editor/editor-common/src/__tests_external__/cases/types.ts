import type { InProductCommonTestCaseOpts } from '@atlaskit/in-product-testing';

export interface EditorTestCaseOpts extends InProductCommonTestCaseOpts {
  ui?: {
    /* DOM selector of the publish button */
    publishButton?: string;
  };

  runOnly?: string[];
}
