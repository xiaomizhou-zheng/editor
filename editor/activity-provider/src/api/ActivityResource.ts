import { utils } from '@atlaskit/util-service-support';
import {
  ActivityItem,
  ActivityResponse,
  ActivityProvider,
  ActivityContainer,
} from '../types';
import { ActivityError } from './error';

export const makeGetRecentItemBody = (cloudId: string) => ({
  query: `
    query editor_recentActivities($filter: [ActivitiesFilter!], $first: Int) {
      activities {
        myActivities {
          viewed(filters: $filter, first: $first) {
            nodes {
              timestamp,
              object {
                id,
                name,
                type,
                url,
                iconUrl,
                containers {
                  name,
                }
              }
            }
          }
        }
      }
    }
  `,
  variables: {
    first: 200,
    filter: [
      {
        type: 'AND',
        arguments: {
          cloudIds: [cloudId],
        },
      },
    ],
  },
});

/**
 * @deprecated - Please use @atlassian/recent-work-client instead
 */
export default class ActivityResource implements ActivityProvider {
  private recentPromise?: Promise<ActivityResponse>;
  private url: string;
  private cloudId: string;
  private options: RequestInit;

  constructor(url: string, cloudId: string, options: RequestInit = {}) {
    this.url = url;
    this.cloudId = cloudId;
    this.options = options;
  }

  public async getRecentItems() {
    if (!this.recentPromise) {
      const options: RequestInit = {
        mode: 'cors',
        method: 'POST',
        cache: 'no-cache',
        credentials: 'same-origin',
        headers: {
          'Content-Type': 'application/json',
        },
        redirect: 'follow',
        referrer: 'no-referrer',
        ...this.options,
        body: JSON.stringify(makeGetRecentItemBody(this.cloudId)),
      };

      this.recentPromise = utils.requestService(
        { url: this.url },
        { requestInit: options },
      );
    }

    try {
      const response = await this.recentPromise;
      return response.data.activities.myActivities.viewed.nodes.map((node) => ({
        objectId: atob(node.object.id),
        name: node.object.name,
        container: this.getContainerName(node.object.containers),
        url: node.object.url,
        iconUrl: node.object.iconUrl,
        type: node.object.type,
        viewedTimestamp: node.timestamp,
      }));
    } catch (err: any) {
      throw new ActivityError(err.reason, err.code);
    }
  }

  public async searchRecent(query: string) {
    const items = await this.getRecentItems();
    return this.filterItems(items, query);
  }

  /**
   * It should return the closet container's name
   */
  private getContainerName(containers: Array<ActivityContainer>): string {
    for (let i = containers.length - 1; i >= 0; --i) {
      const containerName = containers[i].name;
      if (containerName) {
        return containerName;
      }
    }
    return '';
  }

  private filterItems(
    items: Array<ActivityItem>,
    searchTerm: string,
  ): Array<ActivityItem> {
    if (!searchTerm) {
      return [];
    }

    return items.filter((item) => {
      return item.name.toLowerCase().indexOf(searchTerm.toLowerCase()) > -1;
    });
  }
}
