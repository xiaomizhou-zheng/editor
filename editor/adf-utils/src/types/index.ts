export interface ADFEntityMark {
  type: string;
  attrs?: { [name: string]: any };
}

export interface ADFEntity {
  type: string;
  attrs?: { [name: string]: any };
  content?: Array<ADFEntity | undefined>;
  marks?: Array<ADFEntityMark>;
  text?: string;
  [key: string]: any;
}

export type Visitor = (
  node: ADFEntity,
  parent: EntityParent,
  index: number,
  depth: number,
) => ADFEntity | false | undefined | void;

export type VisitorCollection = { [nodeType: string]: Visitor };

export type EntityParent = { node?: ADFEntity; parent?: EntityParent };
