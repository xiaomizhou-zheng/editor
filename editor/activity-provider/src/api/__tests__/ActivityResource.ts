import fetchMock from 'fetch-mock/es5/client';
import { graphql, buildSchema } from 'graphql';
import { promisify } from 'util';
import * as fs from 'fs';
import { join } from 'path';
import ActivityResource, { makeGetRecentItemBody } from '../ActivityResource';
import { ActivityResponse } from '../../types';

const readFile = promisify(fs.readFile);

const activityUrl = '/graphql';
const cloudId = '123';

const getMockRecentResponse = (): ActivityResponse => ({
  data: {
    activities: {
      myActivities: {
        viewed: {
          nodes: [
            {
              id:
                'YXJpOmNsb3VkOmFjdGl2aXR5OkRVTU1ZLTE1OGM4MjA0LWZmM2ItNDdjMi1hZGJiLWEwOTA2Y2NjNzIyYjppdGVtLzE3MzM3MGZjNmJlYWRiNDI1YjYyMTJjZmQ5NjM4YWViZWRkMTUzMWUxZGU=',
              timestamp: '2020-06-24T01:15:39.680Z',
              object: {
                id:
                  'YXJpOmNsb3VkOmNvbmZsdWVuY2U6RFVNTVktMTU4YzgyMDQtZmYzYi00N2MyLWFkYmItYTA5MDZjY2M3MjJiOmNvbnRlbnQvMTQ1NjgzNDI5MQ==',
                localResourceId: '159894',
                name: 'recent item 1',
                type: 'ISSUE',
                product: 'JIRA_SOFTWARE',
                cloudId: 'DUMMY-158c8204-ff3b-47c2-adbb-a0906ccc722b',
                url: 'https://product-fabric.atlassian.net/browse/EDM-642',
                iconUrl:
                  'https://product-fabric.atlassian.net/secure/viewavatar?size=medium&avatarId=10318&avatarType=issuetype',
                containers: [
                  {
                    id:
                      'RFVNTVktMTU4YzgyMDQtZmYzYi00N2MyLWFkYmItYTA5MDZjY2M3MjJi',
                    name: null,
                    type: 'SITE',
                    product: 'JIRA_SOFTWARE',
                  },
                  {
                    id:
                      'YXJpOmNsb3VkOmppcmE6RFVNTVktMTU4YzgyMDQtZmYzYi00N2MyLWFkYmItYTA5MDZjY2M3MjJiOnByb2plY3QvMTMwNTU=',
                    name: 'Editor Media',
                    type: 'PROJECT',
                    product: 'JIRA_SOFTWARE',
                  },
                ],
              },
            },
            {
              id:
                'YXJpOmNsb3VkOmFjdGl2aXR5OkRVTU1ZLTE1OGM4MjA0LWZmM2ItNDdjMi1hZGJiLWEwOTA2Y2NjNzIyYjppdGVtLzE3MmUzZTA5MTNkMWJkYzE5NTgyMTU1ZTc5MmVhZDk2ZGQ4YzM1ZWE5NDc=',
              timestamp: '2020-06-24T01:10:44.797Z',
              object: {
                id:
                  'YXJpOmNsb3VkOmppcmE6RFVNTVktMTU4YzgyMDQtZmYzYi00N2MyLWFkYmItYTA5MDZjY2M3MjJiOmlzc3VlLzE1OTk0MQ==',
                localResourceId: '159941',
                name: 'recent item 2',
                type: 'longIssue' as any,
                product: 'JIRA_SOFTWARE',
                cloudId: 'DUMMY-158c8204-ff3b-47c2-adbb-a0906ccc722b',
                url: 'https://product-fabric.atlassian.net/browse/EDM-649',
                iconUrl:
                  'https://product-fabric.atlassian.net/secure/viewavatar?size=medium&avatarId=10318&avatarType=issuetype',
                containers: [
                  {
                    id:
                      'RFVNTVktMTU4YzgyMDQtZmYzYi00N2MyLWFkYmItYTA5MDZjY2M3MjJi',
                    name: null,
                    type: 'SITE',
                    product: 'JIRA_SOFTWARE',
                  },
                  {
                    id:
                      'YXJpOmNsb3VkOmppcmE6RFVNTVktMTU4YzgyMDQtZmYzYi00N2MyLWFkYmItYTA5MDZjY2M3MjJiOnByb2plY3QvMTMwNTU=',
                    name: '',
                    type: 'PROJECT',
                    product: 'JIRA_SOFTWARE',
                  },
                ],
              },
            },
          ],
        },
      },
    },
  },
});

describe('ActivityResource', () => {
  beforeEach(() => {
    fetchMock.mock({
      matcher: activityUrl,
      response: getMockRecentResponse(),
      name: 'recent',
    });
  });

  afterEach(fetchMock.restore);

  it('should validate against our graphql schema', async () => {
    const schemaFile = await readFile(
      join(__dirname, 'activities.graphql'),
      'utf8',
    );
    const activitySchema = buildSchema(schemaFile, { assumeValid: true });

    const result = await graphql(
      activitySchema,
      makeGetRecentItemBody(cloudId).query,
    );
    expect(result.errors).toBeUndefined();
  });

  it('should resolve recent items', async () => {
    const provider = new ActivityResource(activityUrl, cloudId);
    const items = await provider.getRecentItems();
    expect(items).toMatchSnapshot();
  });

  describe('search recent', () => {
    it('should return empty array with empty search term', async () => {
      const provider = new ActivityResource(activityUrl, cloudId);
      const items = await provider.searchRecent('');
      expect(items).toHaveLength(0);
    });

    it('should return empty array with search term that matches nothing', async () => {
      const provider = new ActivityResource(activityUrl, cloudId);
      const items = await provider.searchRecent('klasjfklajsf');
      expect(items).toHaveLength(0);
    });

    it('should perform case-insensitive prefix search on name property', async () => {
      const provider = new ActivityResource(activityUrl, cloudId);
      const items = await provider.searchRecent('recent');
      expect(items).toMatchSnapshot();
    });
  });
});
