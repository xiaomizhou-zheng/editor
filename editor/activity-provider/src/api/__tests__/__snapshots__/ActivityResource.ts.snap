// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityResource search recent should perform case-insensitive prefix search on name property 1`] = `
Array [
  Object {
    "container": "Editor Media",
    "iconUrl": "https://product-fabric.atlassian.net/secure/viewavatar?size=medium&avatarId=10318&avatarType=issuetype",
    "name": "recent item 1",
    "objectId": "ari:cloud:confluence:DUMMY-158c8204-ff3b-47c2-adbb-a0906ccc722b:content/1456834291",
    "type": "ISSUE",
    "url": "https://product-fabric.atlassian.net/browse/EDM-642",
    "viewedTimestamp": "2020-06-24T01:15:39.680Z",
  },
  Object {
    "container": "",
    "iconUrl": "https://product-fabric.atlassian.net/secure/viewavatar?size=medium&avatarId=10318&avatarType=issuetype",
    "name": "recent item 2",
    "objectId": "ari:cloud:jira:DUMMY-158c8204-ff3b-47c2-adbb-a0906ccc722b:issue/159941",
    "type": "longIssue",
    "url": "https://product-fabric.atlassian.net/browse/EDM-649",
    "viewedTimestamp": "2020-06-24T01:10:44.797Z",
  },
]
`;

exports[`ActivityResource should resolve recent items 1`] = `
Array [
  Object {
    "container": "Editor Media",
    "iconUrl": "https://product-fabric.atlassian.net/secure/viewavatar?size=medium&avatarId=10318&avatarType=issuetype",
    "name": "recent item 1",
    "objectId": "ari:cloud:confluence:DUMMY-158c8204-ff3b-47c2-adbb-a0906ccc722b:content/1456834291",
    "type": "ISSUE",
    "url": "https://product-fabric.atlassian.net/browse/EDM-642",
    "viewedTimestamp": "2020-06-24T01:15:39.680Z",
  },
  Object {
    "container": "",
    "iconUrl": "https://product-fabric.atlassian.net/secure/viewavatar?size=medium&avatarId=10318&avatarType=issuetype",
    "name": "recent item 2",
    "objectId": "ari:cloud:jira:DUMMY-158c8204-ff3b-47c2-adbb-a0906ccc722b:issue/159941",
    "type": "longIssue",
    "url": "https://product-fabric.atlassian.net/browse/EDM-649",
    "viewedTimestamp": "2020-06-24T01:10:44.797Z",
  },
]
`;
